### PR TITLE
servant-client-ghcjs: Support binary requests

### PR DESCRIPTION
**Pull request by FPtje**
Introduces support for both sending and receiving binary data in `servant-client-ghcjs`.

Previously, the conversion to String before going to Bytestring would fail with any requests that sent or received binary data. With ArrayBuffers in the backend, sending Binary encoded data now works.

Also added a test case to https://github.com/FPtje/servant-client-ghcjs-test to test both sending and receiving of Binary data. It works and the tests don't show any regression.

One important note: I see that some recent version of servant has added a `streamingRequest` to the `RunClient`class. It is't implemented in servant-client-ghcjs, and this PR doesn't add it.

### CODE REVIEW

**Comment by alpmestan** [comment]
@arianvp Thanks for offering to look into it. :-)

**Comment by phadej** [comment]
`fail`, not `error`. Even better would be to `throwIO` with custom exception type.

File: servant-client-ghcjs/src/Servant/Client/Internal/XhrClient.hs [hunk 1]
```diff
@@ -217,42 +224,47 @@ setHeaders xhr request = do
...
 
   where
-    go :: RequestBody -> IO (Maybe String)
+    go :: RequestBody -> IO ArrayBuffer
     go x = case x of
-      RequestBodyLBS x -> return $ mBody x
+      RequestBodyLBS x -> return $ mBody $ BL.toStrict x
       RequestBodyBS x -> return $ mBody x
-      RequestBodyBuilder _ x -> return $ mBody $ toLazyByteString x
+      RequestBodyBuilder _ x -> return $ mBody $ BL.toStrict $ toLazyByteString x
```
Comment (phadej) [inline servant-client-ghcjs/src/Servant/Client/Internal/XhrClient.hs]: As we on streaming case, we probably should add a note that `servant-client-ghcjs` doesn't respect any "constant space request body streaming" undertakings. (haddocks of `ClientM` and `client`?)

**Comment by FPtje** [comment]
Updated: Rebased on latest master, added haddocks to `client` and `clientM` mentioning that streaming isn't supported. `streamingRequest` does a `throwIO` with a custom exception.

**Comment by arianvp** [comment]
LGTM @phadej ?

**Comment by arianvp** [comment]
Bump. Is this okay to merge?

**Pull request merged**